### PR TITLE
[ETFE-3284] - Add API for private/public beta eligibility handling

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

**Comment by LewisMitchell** [comment]
App config PR's:
https://github.com/hmrc/app-config-qa/pull/21341
https://github.com/hmrc/app-config-staging/pull/14620
https://github.com/hmrc/app-config-production/pull/21062

**Comment by LewisMitchell** [comment]
rebuild

File: app/uk/gov/hmrc/emcstfe/services/userAllowList/UserAllowListService.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      Future(Right(false))
+    }
+  }
+
+  private[services] def checkPublicBetaEligibility(service: String)(implicit request: UserRequest[_]): Boolean = {
+    // ERN drop first 4 and drop the last 2 and take 2 from the result (which saves performing a modulus calculation)
+    // It adds 1 to prevent returning 0 as a final result so the 'ernAsPercentile' is between 1-100
+    config.publicBetaTrafficPercentageForService(service).exists { trafficPercentage =>
+      request.allUserERNs.map { userErn =>
+        val ernAsPercentile = userErn.drop(4).dropRight(2).takeRight(2)
```
Comment (grahampaulcook) [inline app/uk/gov/hmrc/emcstfe/services/userAllowList/UserAllowListService.scala:58]: I don't think we need to drop(4) as were only interested in the right hand side

Comment (LewisMitchell) [reply app/uk/gov/hmrc/emcstfe/services/userAllowList/UserAllowListService.scala:58]: Good point, will update

**Review by grahampaulcook — APPROVED** [review]

**Pull request merged**